src/foam/core/stdlib.js

```diff
@@ -587,10 +586,8 @@ foam.LIB({
     function getTime(d) { return ! d ? 0 : d.getTime ? d.getTime() : d ; },
     function equals(a, b) { return this.getTime(a) === this.getTime(b); },
     function compare(a, b) {
-      if ( ! foam.Date.isInstance(b) && ! foam.Number.isInstance(b) ) return 1;
-      a = this.getTime(a);
-      b = foam.Number.isInstance(b) ? b : this.getTime(b);
-      return foam.Number.compare(a, b);
+      if ( ! foam.Date.isInstance(b) ) return 1;
```

Comment:
Double checking to make sure this is intentional. This change makes the Date comparator a strict comparison. Previously, you could perform an operation such as
```
var x = new Date();
foam.Date.compare(x, x.getTime()); // Returning a 0, implying equivalence.
```
This was also the reason for the weird ordering of the types originally.